### PR TITLE
 core: provide function to check msg queue initialization

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
I introduced this in my reference implementation for #10399, but I think readability of the code base can benefit from this regardless if we go forward with this issue or not.

### Testing procedure
- `examples/gnrc_networking` should still work without hitting an assertion on initialization
- `tests/gnrc_netif` should still pass
- `tests/msg_*` should still pass

### Issues/PRs references
Introduced in reference implementation for #10399 but factually independent of that issue.

### CODE REVIEW

**Comment by miri64** [comment]
@kaspar030 ping?

File: sys/net/gnrc/netreg/gnrc_netreg.c [hunk 1]
```diff
@@ -39,21 +39,20 @@ void gnrc_netreg_init(void)
...
-    bool has_msg_q = (entry->type != GNRC_NETREG_TYPE_DEFAULT) ||
-                     sched_threads[entry->target.pid]->msg_array;
-#else
-    bool has_msg_q = sched_threads[entry->target.pid]->msg_array;
-#endif
+#if DEVELHELP
+    bool has_msg_q = thread_has_queue(sched_threads[entry->target.pid]);
+
+# if defined(MODULE_GNRC_NETAPI_MBOX) || defined(MODULE_GNRC_NETAPI_CALLBACKS)
+    has_msg_q = (entry->type != GNRC_NETREG_TYPE_DEFAULT) || has_msg_q;
```
Comment (kaspar030) [inline sys/net/gnrc/netreg/gnrc_netreg.c]: It's less readable, but if you move the ```thread_has_queue()``` call after the ||, it'll be executed conditionally.

**Comment by kaspar030** [comment]
Any thoughts on renaming the function to "thread_has_msg_queue()"?

File: sys/net/gnrc/netreg/gnrc_netreg.c (hunk 1 above)
Comment (miri64) [reply sys/net/gnrc/netreg/gnrc_netreg.c]: For `DEVELHELP` code, I rather have it readable than optimized.

Comment (kaspar030) [reply sys/net/gnrc/netreg/gnrc_netreg.c]: ok!

**Comment by kaspar030** [comment]
> Any thoughts on renaming the function to "thread_has_msg_queue()"?

Because, threads are (or are not) in a pending queue, have a waiters priority queue for messages and mutexes, and possibly a message queue.

**Comment by miri64** [comment]
> > Any thoughts on renaming the function to "thread_has_msg_queue()"?
> 
> Because, threads are (or are not) in a pending queue, have a waiters priority queue for messages and mutexes, and possibly a message queue.

Good point!

**Comment by kaspar030** [comment]
ACK. Pls squash!

**Comment by miri64** [comment]
Squashed

**Comment by miri64** [comment]
Should I ask for a second review then, since it is changing `core_msg`?

**Comment by miri64** [comment]
Turns out you were right about the check in `gnrc_netreg`... If the type is not checked beforehand (if different types exists) `target->pid` evaluates to garbage, and the node crashes ;-) (thanks Murdock for detecting this :-))

**Comment by kaspar030** [comment]
@miri64 Do we need to handle core_msg not being compiled in?

**Comment by miri64** [comment]
@kaspar030 do we have an app to test for that?

**Comment by kaspar030** [comment]
> Yes we need to account for that ;-).

Ok. I was wondering if any application actually interested in the result would need to depend on core_msg anyways, but as it is now I think the message is clear. ;)

Please squash, re-ACK!

**Comment by kaspar030** [comment]
> So is `tests/minimal` actually minimal?

```bool so_is_tests/minimal_actually_minimal(void) { return sizeof(tests/minimal)==0; }```

**Comment by miri64** [comment]
You should test when I test something ;-). Found another bug (but squashed that immediately).

**Comment by miri64** [comment]
@kaspar030 good to go?

**Comment by kaspar030** [comment]
yeah! &go.

**Pull request merged**